module-3/lab-supervised-learning-sklearn/your-code/main.ipynb

```diff
@@ -491,7 +491,7 @@
     "diabetes_data_test=X.loc[len(X)-20:]\n",
```

Comment:
- "1. There are 10 attributes in the data. Age, Sex, Body mass index, Average blood pressure are
    quite self-explanatory. S1, S2, S3, S4, S5 and S6 are 6 different blood serum measurements. 
2. The diabetes['data'] is what we are going to know in all cases, the ones we already know and the ones we
    pretend to predict in the future. The diabetes['target'] is the information we know just for the registers
    in our dataset, and what we are trying to predict for future registers.
3. 442 registers"  Your target (y) or the variable you want to predict with your model is the disease progression one year after baseline. The data contains the observed data for each quantitative measure of disease progression, in other words the observed data of every variable and disease progression measure. You would need to assess which of the attributes provided explained the variability of your target. 

- "Interpreting hypothesis testing results
Answer the following questions in the cell below:
What is the F-score of your linear model and is the null hypothesis rejected?
Does any of the t-tests of the coefficients produce a confidence interval containing zero? What are they?
How will you modify your linear reguression model according to the test results above?"

1. The F-statistic is a measure of the significance of the model, the null hypothesis of this test is that the explanatory variables in the model are not effective. That means that for a 95% of confidence level (5% of significance) if the p value drops below 0.05 we can reject the null hypothesis and conclude that the model is statistically significant. This case where Prob (F-statistic): 4.64e-58 we can reject the null hypothesis.
2. The t-statistic is used to assess whether or not each explanatory variable in the model is statistically significant. The null hypothesis is that the coefficient is equal to zero or in other words the variable is not helping the model. For a 95% confidence level we reject the null hypothesis if the p-value drops below 0.05 or in other words the variable is statistically significant.
This case you have x1, x5, x6, x7,x8 and x10 p-values above 0.05 so we can conclude this variables are not statistically significant and const, x2, x3, x4 and x9 p-values below 0.05 so we can conclude these ones are statistically significant.
3. Another important measure you should check when estimating by OLS is the R-Squared and Adjusted R-Squared. Both are measures of model performance. The R-Squared tends to grow as the model complexity grows (as more variables are included) so we usually look at the Adjusted R-Squared because it remains stable whatever the number of variables in the model. Adjusted R-square is lower than R-squared. If you find a big difference between R-Squared and Adjusted R-Squared, you would need to redefine your linear regression.
This case the R-Squared and the Adjusted R-Squared are quite low, generally speaking a model with theses measures below 60% need to be reviewed.
4. Try to run the model again removing non significant variables and check the results.
5. There is one additional stuff you should run in this kind of estimator, residuals analysis. You need to check if the residuals meet the hypothesis of the OLS method.
I have been searching in Internet for some statistic resources on this kind of estimator but no one is very good. However, I include below three links:
http://resources.esri.com/help/9.3/arcgisengine/java/gp_toolref/spatial_statistics_tools/interpreting_ols_results.htm
http://statisticsbyjim.com/regression/ols-linear-regression-assumptions/
https://statinfer.com/204-1-7-adjusted-r-squared-in-python/
I recommend you this page for ML (I don't know if they have any statistics course)
https://trainings.analyticsvidhya.com/
Still the best way of learning statistics is reading books and take some classes. The following one can help you in first steps.